js/privatebin.js

```diff
@@ -391,9 +404,11 @@ jQuery.PrivateBin = (function($, RawDeflate) {
         me.urls2links = function(element)
         {
             element.html(
-                element.html().replace(
-                    /(((https?|ftp):\/\/[\w?!=&.\/-;#@~%+*-]+(?![\w\s?!&.\/;#~%"=-]>))|((magnet):[\w?=&.\/-;#@~%+*-]+))/ig,
-                    '<a href="$1" rel="nofollow">$1</a>'
+                DOMPurify.sanitize(
```

Comment:
Got it, thanks for pointing out the intention behind this.
My change shouldn't cause the magnet links to stop working, as the magnet links had already been converted to anchor elements before call to `DOMPurify.sanitize`

![image](https://user-images.githubusercontent.com/10842569/82135737-bde95f00-9838-11ea-9798-9b4dc1a680dd.png)

Apologize for not having a demo site immediately available, I'm working on setting up one ASAP.

Reply:
What I discovered was that DOMpurify has a very narrow set of URL schemes that it supports - magnet links were broken after we started applying sanitize() after/on top of the url2links() function.

https://github.com/cure53/DOMPurify/blob/master/src/regexp.js#L8
https://github.com/cure53/DOMPurify/blob/master/demos/README.md#hook-to-white-list-safe-uri-schemes-link

Reply:
Thanks looking into it now.

Reply:
Sorry for the inconvenience, I'm unable to reproduce the bug, I guess that is my operating system + browser combination does not exhibit this specific problem.

It seemed even after sanitize, when I click on the magnet link, my TransmissionBT will be able to pick up the magnet link.

I now have the demo up here:
https://privatebin-staging.herokuapp.com/

Would you please kindly provide the error log so I can further debug this problem?
Thank you in advance.

Reply:
I can confirm: Your version doesn't have that issue and magnet links work perfectly fine. Sorry for the noise. For your reference, I used the [sample collecting a number of edge cases](https://github.com/PrivateBin/PrivateBin/issues/588#issuecomment-595399629) that we discovered when we struggled with the encoding problems in 1.3.3:
https://privatebin-staging.herokuapp.com/?60954ea0a2f0b5d5#yMb5uYq6T9RWfSRruF3gfe6P8oSh9M9LGwLynhr3uBk

I am happy that we can have a DOMPurify in that place, in case someone manages to craft a malicious URL that the regex accepts - before your addition this would have been displayed unsanitized.